cypress/e2e/GUI/testGui.cy.js

```diff
@@ -0,0 +1,44 @@
+/// <reference types="Cypress" />
+
+describe('Testando a aplicação EngageSphere', () => {
+
+    // Executa antes de cada teste
+    beforeEach(() => {
+      // Define uma data e hora fixa
+      const fixedDate = new Date(2024, 6, 10); // 10 de julho de 2024
+      cy.clock(fixedDate.getTime());
+      cy.visit('http://localhost:3000');
+    });
+  
+    // Bloco de testes
+    it('Testando o componente Header', () => {
+      cy.get('h1').should('have.text', 'EngageSphere');
+      cy.contains('p', 'Hi there! It is now Wed Jul 10 2024.')
+    
+    });
+  
+    it('Testando o componente Input', () => {
+      cy.get('[data-testid="name"]').type('Joe');
+      cy.get('[data-testid="table"] > :nth-child(1)').should('have.text', 'Hi Joe! It is now Wed Jul 10 2024.');
```

Comment:
Este seletor não é descritivo. Veja uma alternativa abaixo.

```suggestion
      cy.contains('p', 'Hi Joe! It is now Wed Jul 10 2024.').should('be.visible');
```

Perceba também a _assertion_ (`.should('be.visible')`)